components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt

```diff
@@ -43,6 +47,13 @@ class UtxoFinalityFlowV1(
 
     private val transactionId = initialTransaction.id
 
+    /*
+    * only if the number of sessions(counterparties) is more than one,
+    * it should wait for additional signatures.
+    * Otherwise, it can be skipped since there isn't unseen signatures
+    */
+    private val waitForAdditionalSignatures = !(version == UtxoFinalityVersion.V2 && sessions.size == 1)
```

Comment:
```suggestion
    private val waitForAdditionalSignatures = (version == UtxoFinalityVersion.V1 || sessions.size >1)
```
looks a bit simpler for me.